tags/tags.py

```diff
@@ -0,0 +1,265 @@
+from datetime import datetime
+
+import discord
+from discord import Guild
+from redbot.core import Config
+from redbot.core.bot import Red
+from redbot.core import commands
+from redbot.core.utils.mod import is_mod_or_superior
+
+
+class TagCog(commands.Cog):
+    """Tag cog"""
+
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.config = Config.get_conf(self, identifier=128986274420752384002)
+
+        default_guild_config = {
+            "log": {  # TODO: The log booleans are not respected right now. Everything is logged.
+                "transfers": True,
+                "uses": True
+            },
+            "tags": {
+                "example": {
+                    "creator": "128986274420752384",
+                    "owner": "128986274420752384",
+                    "created": int(datetime.utcnow().timestamp()),
+                    "content": "This is an example tag!",
+                    "transfers": [
+                        {
+                            "time": int(datetime.utcnow().timestamp()),
+                            "from": "128986274420752384",
+                            "to": "128986274420752384"
+                        }
+                    ],
+                    "uses": [
+                        {
+                            "user": "128986274420752384",
+                            "time": int(datetime.utcnow().timestamp())
+                        }
+                    ]
+                }
+            },
+            "aliases": {
+                "ex": {
+                    "tag": "example",
+                    "creator": "128986274420752384",
+                    "created": int(datetime.utcnow().timestamp()),
+                    "uses": [
+                        {
+                            "user": "128986274420752384",
+                            "time": int(datetime.utcnow().timestamp())
+                        }
+                    ]
+                }
+            }
+        }
+
+        self.config.register_guild(**default_guild_config)
+
+    @commands.guild_only()
+    @commands.group(name="tag", pass_context=True, invoke_without_command=True)
+    async def _tag(self, ctx: commands.Context, tag: str):
+
+        async def fire_tag(t) -> bool:
```

Comment:
Not a big fan of this function being nested. Maybe raise it to a class method?

Reply:
Done!